backend/src/walk/walk.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { WalkService } from './walk.service';
+
+import { Dogs } from '../dogs/dogs.entity';
+import { DogsService } from '../dogs/dogs.service';
+import { DogSummary } from '../dogs/types/dog-summary.type';
+import { UsersService } from '../users/users.service';
+
+describe('WalkService', () => {
+    let service: WalkService;
+    let usersService: UsersService;
+    let dogsService: DogsService;
+    let dogRepository: Repository<Dogs>;
+
+    beforeEach(async () => {
+        const module: TestingModule = await Test.createTestingModule({
+            providers: [
+                WalkService,
+                UsersService,
+                {
+                    provide: UsersService,
+                    useValue: {
+                        getOwnDogsList: jest.fn(),
+                    },
+                },
+                {
+                    provide: DogsService,
+                    useValue: {
+                        getDogsSummaryList: jest.fn(),
+                        findOne: jest.fn(),
+                        updateIsWalking: jest.fn(),
+                    },
+                },
+                {
+                    provide: getRepositoryToken(Dogs),
+                    useClass: Repository,
+                },
+            ],
+        }).compile();
+
+        service = module.get<WalkService>(WalkService);
+        usersService = module.get<UsersService>(UsersService);
+        dogsService = module.get<DogsService>(DogsService);
+        dogRepository = module.get<Repository<Dogs>>(getRepositoryToken(Dogs));
+    });
+
+    describe('getAvailableDogs', () => {
+        const userId = 1;
+        const ownDogIds = [1, 2, 3];
+        const availableDogs: DogSummary[] = [
+            { id: 1, name: 'dangdang1', profilePhotoUrl: 'https://example.com/dog1.jpg' },
+            { id: 2, name: 'dangdang2', profilePhotoUrl: 'https://example.com/dog2.jpg' },
+        ];
+
+        context('사용자 id가 주어지면', () => {
+            beforeEach(() => {
+                jest.spyOn(usersService, 'getOwnDogsList').mockResolvedValueOnce(ownDogIds);
+                jest.spyOn(service, 'checkAvailableDogs').mockResolvedValueOnce();
+                jest.spyOn(dogsService, 'getDogsSummaryList').mockResolvedValue(availableDogs);
+            });
+
+            it('소유한 강아지 목록을 반환한다.', async () => {
+                const dogSummaries = await service.getAvailableDogs(userId);
+
+                expect(dogSummaries).toEqual([
+                    { id: 1, name: 'dangdang1', profilePhotoUrl: 'https://example.com/dog1.jpg' },
+                    { id: 2, name: 'dangdang2', profilePhotoUrl: 'https://example.com/dog2.jpg' },
+                ]);
```

Comment:
```suggestion
                expect(dogSummaries).toEqual(availableDogs);
```
expect로 결과 값 검증할 때 이렇게 쓰는 것보다 값을 직접 적어주는게 좋은가요?

Reply:
직접 값을 표시한 이유는 테스트 결과를 명시적으로 확인 할 수 있어서 `availableDogs`를 사용하지 않고 직접 값을 사용했습니다.